### PR TITLE
feat(fmt): supports color module

### PR DESCRIPTION
**Pull request by lino-levan**
This PR is part one of completing https://github.com/denoland/deno_std/issues/2895.

This was heavily inspired `chalk/supports-color` but the api was tweaked a little bit to fit a little bit better into typescript. The tests were also taken from `chalk/supports-color` but also heavily modified.

Part two (which would be in another PR) would be:
> When the client tries to use a color method in a non-color terminal have it do nothing

This is a change that I'm not entirely sold on. I think it would be better for DX but it might be a little too "magic" for a standard library.

### CODE REVIEW

**Comment by lino-levan** [comment]
Can't reproc test failures in local env. Will play around with GH actions.

**Comment by lino-levan** [comment]
It turns out that this is just an issue with the tests that `chalk/supports-color` has. They do not do CI testing on windows and the tests thus fail on windows (even if the result of the modules is correct). I will modify the affected tests.

**Comment by lino-levan** [comment]
The current test failures are unrelated to this change. Looks like something in canary broke std.

File: fmt/supports_color.ts [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+  {
+    streamIsTTY,
+    sniffFlags = true,
+    args = Deno.args,
+    os = Deno.build.os,
+    osRelease = Deno.osRelease(),
+  }: SupportsColorOptions = {},
+) {
+  // Read flags to see if anything matches to force colors or to force no colors
+  const { _, color, colors } = parse(args, {
```
Comment (kt3k) [inline fmt/supports_color.ts:72]: I think we shouldn't automatically read from cli flags

**Comment by kt3k** [comment]
This seems introducing too many env permission prompts to cli tools. Also the dependencies to CI vendor specific env vars look concerning. I feel this feature better belong to 3rd party tools, rather than std.

**Comment by lino-levan** [comment]
SGTM, closing this PR

**Comment by lino-levan** [comment]
For anyone curious about this in the future, I just moved the code to a /x/ module: https://deno.land/x/supports_color

**Pull request closed without merging**